builtin/aws/ecs/releaser.go

```diff
@@ -37,8 +39,9 @@ func (r *Releaser) Release(
 	ui terminal.UI,
 	target *Deployment,
 ) (*Release, error) {
-	if target.LoadBalancerArn == "" && target.TargetGroupArn == "" {
```

Comment:
Here's the first part of the bug. If you specified a listener arn, you'd get here with no LoadBalancerArn, but because you had a target group, you'd pass this check.